docs/site/DataSource-generator.md

```diff
@@ -44,6 +44,17 @@ The tool will prompt you for:
 - **Connector configuration.** If the connector is not a custom connector, the
   CLI will prompt for the connector configuration information.
 
+In case of "In-memory db" connector, you will be prompted for these additional
+details:
+
+- **window.localStorage key** The name of the key for persistence in the
+  browser.
+- **Path to file for persistence** The path of the file where the connector
+  should store its data. If no path is specified, a `.db.json` file derived from
+  the name of the connector will be created in the application directory. If a
+  file already exists with such a name, a random `.db.json` file will be created
+  in the application directory.
```

Comment:
> In LB4, if no name is specified, the generator will generate one based on the name of the connector and create the file in the application directory

I consider this as incorrect behavior. As I see it, the `memory` connector should be primarily used for in-memory storage that does not survive app restarts. This is intentional, as it simplifies scenarios where the app (or perhaps a test suite) wants to start with a clean database.

IMO, we should fix `lb4 datasource` to behave the same way as `lb datasource`, i.e. set "file" to an empty value (or leave it out completely).

> In LB3, the behavior is "unknown" and undocumented.

The behavior is well-know and documented. Quoting from https://loopback.io/doc/en/lb3/Memory-connector.html:

> (...) the memory connector does not persist data after an application stops.